### PR TITLE
Fix test_models.py

### PR DESCRIPTION
**Pull request by sgiardie**
`test_models.py` fails because of a bug in the [eval](https://github.com/simonsobs/fgspectra/blob/cbb5e656a9e9df61c283ab0eaa44b354b7b9e033/fgspectra/cross.py#L120) of FactorizedCrossSpectra. Fixing [line 135](https://github.com/simonsobs/fgspectra/blob/cbb5e656a9e9df61c283ab0eaa44b354b7b9e033/fgspectra/cross.py#L135) as
`f_nu = self._sed(**sed_kwargs)[..., np.newaxis]`
seems to fix the problem with the test. Thanks @HTJense for finding the solution.
See [PR 39](https://github.com/simonsobs/fgspectra/pull/39) for more details.

### CODE REVIEW

**Comment by sgiardie** [comment]
I just found out that this change breaks the construction of a [FG dictionary in MFLike](https://github.com/simonsobs/LAT_MFLike/blob/7b8fffc3fc1b206ab0b1a0122b8478bee106d53f/mflike/theoryforge_MFLike.py#L323), giving error 
```
        fg_dict = {}
        for c1, f1 in enumerate(freqs):
            for c2, f2 in enumerate(freqs):
                for s in requested_cls:
                    fg_dict[s, "all", f1, f2] = np.zeros(len(ell))
                    for comp in self.fg_component_list[s]:
                        if comp == "tSZ_and_CIB":
                            fg_dict[s, "tSZ", f1, f2] = model[s, "tSZ"][c1, c2]
                            fg_dict[s, "cibc", f1, f2] = model[s, "cibc"][c1, c2]
                            fg_dict[s, "tSZxCIB", f1, f2] = (
                                model[s, comp][c1, c2]
                                - model[s, "tSZ"][c1, c2]
                                - model[s, "cibc"][c1, c2]
                            )
                            fg_dict[s, "all", f1, f2] += model[s, comp][c1, c2]
                        else:
                            fg_dict[s, comp, f1, f2] = model[s, comp][c1, c2]
>                           fg_dict[s, "all", f1, f2] += fg_dict[s, comp, f1, f2]
E                           ValueError: non-broadcastable output operand with shape (9001,) doesn't match the broadcast shape (1,9001)
```
I found a quick work around by modifying the eval of FactorizedCrossSpectrum as:
```
        f_nu = self._sed(**sed_kwargs)
        cl = self._cl(**cl_kwargs)
        if f_nu.shape[0] != cl.shape[-1] or (f_nu.shape[0] == 1 and cl.shape[-1] == 1):
            f_nu = f_nu[np.newaxis]

        return np.einsum("l...i,l...j,...l->...ijl", f_nu, f_nu, cl)
```
(since we had problem in the test when f_nu.shape[0] == cl.shape[-1] == 1)
it works for now, but I'm not sure that's the best solution

File: fgspectra/cross.py [hunk 1]
```diff
@@ -134,7 +134,7 @@ def eval(self, sed_kwargs={}, cl_kwargs={}):
         """
         f_nu = self._sed(**sed_kwargs)
         cl = self._cl(**cl_kwargs)
-        if f_nu.shape[0] != cl.shape[-1]:
+        if f_nu.shape[0] != cl.shape[-1] or (f_nu.shape[0] == 1 and cl.shape[-1] == 1):
```
Comment (xzackli) [inline fgspectra/cross.py:137]: This looks good to me.

File: tests/test_models.py [hunk 2]
```diff
@@ -57,9 +57,9 @@ def test_ACT_models():
         "f0_dust": 149.7,
     }
 
-    fsz = np.array([par["f0_sz"], par["f0_sz"]])
-    fsynch = np.array([par["f0_synch"], par["f0_synch"]])
-    fdust = np.array([par["f0_dust"], par["f0_dust"]])
+    fsz = np.array([par["f0_sz"]])
+    fsynch = np.array([par["f0_synch"]])
+    fdust = np.array([par["f0_dust"]])
```
Comment (xzackli) [inline tests/test_models.py:62]: I'm glad we can revert this change to the test. It'll be more clear now.

**Review by xzackli — APPROVED** [review]
